### ISSUE TITLE
Add optional argument to control PHP memory limit

### ISSUE THREAD

**Issue by cmpaul** [Qafoo/QualityAnalyzer#26]
For larger code bases, it's easy to run out of memory when running QualityAnalyzer handlers.

**Comment by kore** [comment]
Thanks for the issue & pull request – this is a very valid point we should tackle.

I do not really like the idea of creating that many options. If somebody else wants to pass additional php config options we will end up with `count($configOptions) * count($tools)` options, which will make the help output and the code hard to read.

What I can think of is something like mimicking PHPs CLI by allowing multiple `-d foo[=bar]` options which are then passed on to every tool. This is also consistent with what PHPUnit does to set PHP config options.

What do you think? Is there really a need to configure different memory limits per tool?

**Comment by cmpaul** [comment]
Agreed, I didn't love this either. I initially started with one memory_limit param I passed to all analyzers, but that felt a little too clumsy. This was just to fix it quickly due to a time constraint I'm under, but I will definitely take a stab at using the multiple argument option and make another PR -- thanks for the tip!

**Comment by cmpaul** [comment]
@kore - Another thought: I was investigating whether the analyzers could be run in parallel to speed things up. Would this be useful/beneficial to others or just me? If you like this idea, it might be good to have separate memory options per tool to avoid overwhelming the system.

**Comment by kore** [comment]
Parallel running might be a thing one wants to do, but there are other tools which can already accomplish things like this. It is really easy to set up with ant, for example. And this way you can also easily provide your custom tool configuration for each tool.

QafooAnalyzer can already take existing result files and skip running the analyzers itself. I think for more specific running environments users should fall back to that (just use any build tool) and we should not make QafooAnalyzer another one. I'd prefer to focus on visualization, etc.

**Comment by cmpaul** [comment]
Makes perfect sense, thanks @kore.

**Comment by kore** [comment]
Sorry for not getting back to you earlier.

I thought about the PR and I still feel the handler execution (anticipating similar future extensions based on this) will get too complex this way.

I added two example files and a README section showing how to execute the tools manually, which will allow you to configure this easily and fine grained: https://github.com/Qafoo/QualityAnalyzer#customize-tools

I hope this is a valid compromise for you between simplicity of the software and customization of the tool execution.
